src/ui/Facet/Facet.ts

```diff
@@ -1178,6 +1178,15 @@ export class Facet extends Component {
     $$(this.moreElement).addClass('coveo-active');
     this.values.sortValuesDependingOnStatus(this.numberOfValues);
     this.rebuildValueElements();
+    this.usageAnalytics.logCustomEvent<IAnalyticsFacetMeta>(
+      analyticsActionCauseList.facetShowLess,
+      {
```

Comment:
I would create a method to return this object and use it here and on line 1876 (that's a big file!). e.g.

```
private getShowMoreLessAnalyticsArgs() {
    return {
        facetId: this.options.id,
        facetField: this.options.field.toString(),
        facetTitle: this.options.title
    }
}
```

Maybe if we make the method `public`, then you could also reuse it in the unit tests? What do you think @olamothe?

Reply:
I would be a bit against making a method public just for the sake of unit testing it, but if we have a better reason why not

Reply:
Talked with @olamothe I will create the method but it will be private